atcoder-problems-frontend/src/utils/StreakCounter.ts

```diff
@@ -111,3 +111,43 @@ export const countTeeByDate = (
     .map(([dateLabel, count]) => ({ dateLabel, count }))
     .sort((a, b) => a.dateLabel.localeCompare(b.dateLabel));
 };
+
+export const countTeeInTheOldDays = (
+  dailyTeeCount: {
+    dateLabel: string;
+    count: number;
+  }[]
+) => {
+  const DURATION = 30;
+
+  const minDateLabel = dailyTeeCount[0].dateLabel;
+  const maxDateLabel = dailyTeeCount[dailyTeeCount.length - 1].dateLabel;
+  const dateDelta =
+    (+new Date(maxDateLabel) - +new Date(minDateLabel)) / 1000 / 86400;
```

Comment:
Date - Date が millisec になるの読みにくいので、分かりやすさ重視で getTime を使ってはどうでしょう？

```suggestion
    ((new Date(maxDateLabel)).getTime() - (new Date(minDateLabel)).getTime()) / 1000 / 86400;
```